File: src/main/java/org/gecko/tools/VariableBlockCreatorTool.java
package org.gecko.tools;

import javafx.geometry.Point2D;
import javafx.scene.Node;
import org.gecko.actions.Action;
import org.gecko.actions.ActionManager;

public class VariableBlockCreatorTool extends Tool {

    private static final String NAME = "Variable Block Creator Tool";

    public VariableBlockCreatorTool(ActionManager actionManager) {
        super(actionManager);
    }

    @Override
    public String getName() {
        return NAME;
    }

    @Override
    public String getIconPath() {
        //TODO stub
        return null;
    }

    @Override
    public void visitView(Node view) {
        super.visitView(view);
        view.setOnMouseClicked(event -> {
            Point2D position = new Point2D(event.getX(), event.getY());
            Action createVariableBlockAction = actionManager.getActionFactory().createCreateVariableAction(position);
            actionManager.run(createVariableBlockAction);
        });
    }
}

File: src/main/kotlin/org/gecko/tools/VariableBlockCreatorTool.kt
package org.gecko.tools

import javafx.event.EventHandler
import javafx.scene.Cursor
import javafx.scene.input.MouseButton
import javafx.scene.input.MouseEvent
import org.gecko.actions.Action
import org.gecko.actions.ActionManager
import org.gecko.view.views.ViewElementPane

/**
 * A concrete representation of a variable-block-creating-[Tool], utilized for creating a
 * [PortViewModel][org.gecko.viewmodel.Port].
 */
class VariableBlockCreatorTool(actionManager: ActionManager) :
    Tool(actionManager, ToolType.VARIABLE_BLOCK_CREATOR, false) {
    override fun visitView(pane: ViewElementPane) {
        super.visitView(pane)
        pane.draw().cursor = Cursor.CROSSHAIR
        pane.draw().onMouseClicked = EventHandler<MouseEvent> { event: MouseEvent ->
            if (event.button != MouseButton.PRIMARY) {
                return@EventHandler
            }
            val position = pane.screenToWorldCoordinates(event.screenX, event.screenY)
            val createVariableBlockAction: Action = actionManager.actionFactory.createVariable(position)
            actionManager.run(createVariableBlockAction)
        }
    }
}
